File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const result = [];
    let cnt = 0;
    let day = 0;
    let idx = 0;
    
    while (idx < progresses.length) {
        const speed = speeds[idx];
        const progress = progresses[idx] + (speed * day);
        
        if (progress >= 100) {
            cnt++;
            idx++;
            continue;
        }
        
        if (cnt > 0) {
            result.push(cnt);
            cnt = 0;
        }
        
        day += Math.ceil((100 - progress) / speed);
    }
    
    result.push(cnt);
    
    return result;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.kt
class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): List<Int> {
        val answer = mutableListOf<Int>()
        var days = 0
        var cnt = 0
        
        for ((index, progress) in progresses.withIndex()) {
            val speed = speeds[index]
            val remainProgress = 100 - progress - (speed * days)
                
            if (remainProgress > 0) {
                if (cnt > 0) {
                    answer.add(cnt)
                }
                cnt = 0
                days += ((remainProgress - 1) / speed) + 1
            }

            cnt++
        }
        
        answer.add(cnt)
        
        return answer
    }
}